enterprise/suppress-user/internal/badgerdb/badgerdb.go

```diff
@@ -0,0 +1,279 @@
+		maxGoroutines: 1,
+		maxSeedWait:   10 * time.Second,
+	}
+	for _, opt := range opts {
+		opt(b)
+	}
+
+	return b, b.start()
+}
+
+// GetToken returns the current token
+func (b *Repository) GetToken() ([]byte, error) {
+	b.restoringLock.RLock()
+	defer b.restoringLock.RUnlock() // release the read lock at the end of the operation
+	if b.restoring {
+		return nil, model.ErrRestoring
+	}
+
+	var token []byte
+	err := b.db.View(func(txn *badger.Txn) error {
+		item, err := txn.Get([]byte(tokenKey))
+		if err != nil {
+			return err
+		}
+		err = item.Value(func(val []byte) error {
+			token = val
+			return nil
+		})
+		return err
+	})
+	if err != nil && !errors.Is(err, badger.ErrKeyNotFound) {
+		return nil, err
+	}
+	return token, nil
+}
+
+// Suppress returns true if the given user is suppressed, false otherwise
+func (b *Repository) Suppress(workspaceID, userID, sourceID string) (bool, error) {
+	b.restoringLock.RLock()
+	defer b.restoringLock.RUnlock()
+	if b.restoring {
+		return false, model.ErrRestoring
+	}
+
+	keyPrefix := fmt.Sprintf("%s:%s:", workspaceID, userID)
+	err := b.db.View(func(txn *badger.Txn) error {
+		_, err := txn.Get([]byte(keyPrefix + model.Wildcard))
+		if err != badger.ErrKeyNotFound {
+			return err
+		}
+		_, err = txn.Get([]byte(keyPrefix + sourceID))
+		return err
+	})
+	if err != nil {
+		if err == badger.ErrKeyNotFound {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// Add adds the given suppressions to the repository
+func (b *Repository) Add(suppressions []model.Suppression, token []byte) error {
+	b.restoringLock.RLock()
+	defer b.restoringLock.RUnlock()
+	if b.restoring {
+		return model.ErrRestoring
+	}
+	if b.db.IsClosed() {
+		return badger.ErrDBClosed
+	}
+	wb := b.db.NewWriteBatch()
+	defer wb.Cancel()
+
+	for i := range suppressions {
+		suppression := suppressions[i]
+		keyPrefix := fmt.Sprintf("%s:%s:", suppression.WorkspaceID, suppression.UserID)
+		var keys []string
+		if len(suppression.SourceIDs) == 0 {
+			keys = []string{keyPrefix + model.Wildcard}
+		} else {
+			keys = make([]string, len(suppression.SourceIDs))
+			for i, sourceID := range suppression.SourceIDs {
+				keys[i] = keyPrefix + sourceID
+			}
+		}
+		for _, key := range keys {
+			var err error
+			if suppression.Canceled {
+				err = wb.Delete([]byte(key))
+			} else {
+				err = wb.Set([]byte(key), []byte(""))
+			}
+			if err != nil {
+				return err
+			}
+		}
+
+	}
+	if err := wb.Set([]byte(tokenKey), token); err != nil {
+		return err
+	}
+	err := wb.Flush()
+	return err
+}
+
+// start starts the repository
+func (b *Repository) start() error {
+	b.closed = make(chan struct{})
+	var seeder io.Reader
+	if _, err := os.Stat(b.path); os.IsNotExist(err) && b.seederSource != nil {
+		if seeder, err = b.seederSource(); err != nil {
+			return err
+		}
+	}
+
+	opts := badger.
+		DefaultOptions(b.path).
+		WithLogger(blogger{b.log}).
+		WithCompression(options.None).
+		WithIndexCacheSize(16 << 20).
```

Comment:
Isn't it more readable to write the number since it isn't even that big? At least for me it's better than having to think of `16 * 2²º`.

Reply:
This is the pattern to badgerdb's [documentation](https://dgraph.io/docs/badger/get-started/#memory-usage:~:text=opts.IndexCache%20%3D%20100%20%3C%3C%2020) promotes, it seemed a bit weird at first to me too, but get used to it quickly afterwards.

For example, to set a 100 Mb cache:
```
opts.IndexCache = 100 << 20 // 100 mb or some other size based on the amount of data
```

Reply:
```suggestion
		WithIndexCacheSize(16 << 20). // 16mb
```

:stuck_out_tongue: